File: BACKEND/Siscon_API_Backend/Siscon_API_Backend/Models/Morador.cs
namespace Siscon_API_Backend.Models
{
    public class Morador
    {
        public long Id { get; set; }
        public string NumeroDocumento { get; set; }
        public long? TipoDocumentoId { get; set; }
        public string Nome { get; set; }
        public DateTime? DataNascimento { get; set; }
        public int? Idade { get; set; }
        public int? SexoId { get; set; }
        public string Telefone { get; set; }
        public string Email { get; set; }

        public virtual Sexo Sexo { get; set; }
        public virtual TipoDocumento TipoDocumento { get; set; }
    }
}

File: FRONTEND/SisconFrontEnd/SisconFrontEnd/src/app/models/Morador.ts
import { Sexo } from "./Sexo";
import { TipoDocumento } from "./TipoDocumento";

export class Morador {
  id: number;
  numeroDocumento: string;
  tipoDocumentoId: number | null;
  nome: string;
  dataNascimento: Date | null;
  idade: number | null;
  sexoId: number | null;
  telefone: string;
  email: string;
  sexo: Sexo | null;
  tipoDocumento: TipoDocumento | null;
}

export class MoradorPayload {
  data: Morador[];
}

export class MoradorPayloadUniqueResult {
  data: Morador;
}
